### PR TITLE
Resolves #604 - Adding an external links menu to utilities

### LINKED ISSUE #604: Robert: building an External Resources script since today and maybe before
@RoberT-1000 is working on this today, but I need an issue to tag in a Power Pad pull request, so I am making the issue on his behalf.

He's creating an "External Resources" script which counties can modify, to allow counties to specify county-specific favorite links for staff to have access to.

As I want to pull in changes for #601 as well as the installer changes (which really, really need testing), I am going to quick create dummy redirect files. @RoberT-1000, at your leisure, I'd appreciate you uploading your version to the utilities folder in master. :)

### PR DESCRIPTION
**Pull request by RoberT-1000**
BLIP: This script gives beta_users the ability to manage a list of external links important to the work of the front-line staff in that agency.

### CODE REVIEW

File: utilities/external-links.vbs [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+		ELSE 
+			no_file_exists = MsgBox ("A resource file is required for this script to run. Press OK to continue to provide information on resources for your agency. Press CANCEL to stop this script.", vbOKCancel)
+			IF no_file_exists = vbCancel THEN script_end_procedure("Script cancelled.")
+			CALL create_external_resource(default_directory)
+		END IF			
+	END WITH	
+
+	'splitting the contents of the text file into an array
+	external_resources_array = split(external_resources_array, vbNewLine)
+	ReDim ext_res_multi_array(0, 5)
```
Comment (theVKC) [inline utilities/external-links.vbs]: Excellent documentation. :)

File: utilities/external-links.vbs [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+	    	Text 10, 15, 65, 10, "Resource Name:"
+	    	Text 10, 35, 65, 10, "Resource URL:"
+	    	Text 10, 55, 75, 10, "Resource Description:"
+	    EndDialog		
+		
+		DO
+			err_msg = ""
+			Dialog
+				IF ButtonPressed = test_link_button THEN  			'opening the link 
+					Set objExplorer = CreateObject("InternetExplorer.Application")
```
Comment (theVKC) [inline utilities/external-links.vbs]: Do we want to force IE on this?

Idea: we could allow the "big three" to be indicated in the dialog. I know a few agencies who access certain links using specific browsers...

Comment (theVKC) [inline utilities/external-links.vbs]: In any case, personally I'd rather it use the system default, unless you have good reason for using IE in the test area (also, a couple of counties are testing Win10, and I worry about Edge compatibility for this, but we could probably find a tester or two).

File: utilities/external-links.vbs [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+'===============The script============
+'=====================================
+'stopping the script if the agency does not have the default_directory value set...
+IF default_directory = "" THEN script_end_procedure("Your agency's Global Variables file is not properly configured to use this script. You must provide a value for the variable ''default_directory'' for the script to run. Please contact a scripts administrator to resolve this issue.")
+
+'grabbing the user's ID to determine if they should be able to update the links
+Set objNet = CreateObject("WScript.NetWork") 
+windows_user_ID = UCASE(objNet.UserName)
+
+IF InStr(beta_users, windows_user_ID) <> 0 THEN
```
Comment (theVKC) [inline utilities/external-links.vbs]: Do you think we want to use all beta_users for this? I'm open to allowing (at some point, I wouldn't change the script) a new array for "script_admins" which could also open other utilities to the user...

Comment (RoberT-1000) [inline utilities/external-links.vbs]: beta_users is the only list of techy users we currently have in GlobVar. I am also thinking this is not the best plan long term, but for now, I think it's good enough. I definitely think something like script_admins should be added to GlobVar and then that group is responsible for maintaining external links.

File: utilities/external-links.vbs (hunk 1 above)
Comment (RoberT-1000) [inline utilities/external-links.vbs]: Pro Tip for other script writers >> good comments also help you keep track of your code. This thing is a twisted amalgam of custom functions.

Also pro tip >> use the word "gobbins" as frequently as possible.

File: utilities/external-links.vbs (hunk 2 above)
Comment (theVKC) [inline utilities/external-links.vbs]: I have lots of tricks up my sleeve. :)

As far as this is concerned, however, I was thinking a few GlobVars:

```
'------ This part sets browsers up for the External Resources script. Leave all blank to force IE. Otherwise, any browsers indicated will appear as options for admins to specify in the script.

'Indicate the Firefox path. Leave blank to disable Firefox.
firefox_path = "C:\Program Files\Firefox maybe\firefox.exe"

'Indicate the Chrome path. Leave blank to disable Chrome.
chrome_path = "C:\Program Files\Google probably\chrome.exe"

'--------- end browser gobbins
```

This could either go in GlobVar or even in the file what contains all the links.

**Comment by theVKC** [comment]
General comment: I am going to test this one before I pull it in. But my glance evaluation appears to show it's awesome. @RoberT-1000, if you want to add that browser functionality, let me know and I'll hold up on the merge.

**Comment by RoberT-1000** [comment]
Don't hold up the merge for that. Thinking about what needs to be added and how that should be structured, I'm thinking two new variables need to be added to GlobVar... `firefox_file_path` and `chrome_file_path` and then the m-d array for the script needs to be expanded to allow a default browser for each link (some sites are picky)


image

**Comment by theVKC** [comment]
Good deal. Whenever you get time, I'd love to incorporate default browser options.

**Comment by theVKC** [comment]
Shoot! Can you update the script list?

**Comment by RoberT-1000** [comment]
Yep!

**Comment by RoberT-1000** [comment]
Should be good to go now

**Comment by theVKC** [comment]
Shoot. In master we have a new list that conforms with the GitHub guidelines and aligns with the terms used in MAXIS...

**Comment by RoberT-1000** [comment]
Did I update the wrong file?

**Comment by RoberT-1000** [comment]
I totally did

**Comment by theVKC** [comment]
Easy mistake to make when I left both copies on GitHub.

**Comment by RoberT-1000** [comment]
Correct file updated

**Comment by theVKC** [comment]
Ack!!

In the Power Pad, I created redirects for "External **Resources**" and not "External **Links**". And that file has been updated in the list of scripts.

I should have caught that. Can you repull into that file?

#ImSorryRobert

**Comment by RoberT-1000** [comment]
I can...I didn't overwrite that file because you had already put code in that one...I assumed it was for something else.

**Comment by theVKC** [comment]
#608 is the droid I was looking for. Thanks @RoberT-1000!
